app/services/journeys/params_validator.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+module Journeys
```

Comment:
Some of this code feels a little superfluous - the `ParamsValidator` pattern is used elsewhere to e.g. validate parseable dates which I think is useful to include. But the presence of specific attributes seems to replicate most of the logic in the controller where specific list of params are defined for each controller action. I would suspect some of this code could instead be moved to `params.require` checks in the controller instead? Just a thought and may not be something you want to change.

Reply:
yes, perhaps, 🤔 
I will think about moving some of those `presence: true` validations back to the controller as `params.require`

Reply:
..or perhaps you can make more assumptions in the controller about the list of params present knowing that the validator has passed. Fundamentally it would be good to define the list of required/permitted params in one place as much as possible.

Reply:
now updated to use require() in the controller 👍